data_preparation/predictor_data_prep.r

```diff
@@ -245,205 +314,142 @@ process_indicators <- function(crop, region, start_year, end_year, crop_mask_fil
-          } else if ((indicator == "ET0") | (indicator_source == "GLDAS")) {
-            # check data_preparation/global_ETo_FAO
-            rast_stack[rast_stack == -9999] <- NA
-          } else if (grepl("Temperature", indicator, fixed=TRUE)) {
-            # temp_C = temp_K - 273.15
-            rast_stack <- rast_stack - 273.15
-          }
-
-          # replicate crop mask to match number of indicator layers
-          # we set weights for NA values to 0
-          crop_masks = rep(crop_mask, nlyr(rast_stack))
-          crop_masks[is.na(rast_stack)] = 0
-
-          # NOTE the order of multiplication matters
-          rast_stack = rast_stack * crop_masks
-
-          result = extract(rast_stack, sel_shapes, fun=sum, na.rm=TRUE, ID=FALSE)
-          result$adm_id = sel_shapes$adm_id
-
-          result_w = extract(crop_masks, sel_shapes, fun=sum, na.rm=TRUE, ID=FALSE)
-          result_w$adm_id = sel_shapes$adm_id
-
-          # Divide the result by weights. Added a tiny number to avoid division by Zero.
-          result[-ncol(result)] = result[-ncol(result)]/(result_w[-ncol(result_w)] + 1e-6)
-          result <- melt(result)
-          
-          colnames(result) <- c("adm_id", "ind_file", indicator)
-          result$date <- str_sub(result$ind_file, -8,-1)
-          result$crop_name <- crop
-          result <- result[, c("crop_name", "adm_id",
-                               "date", indicator)]
-          # print(head(result))
-          if (is.null(region_results)) {
-            region_results <- result
-          } else {
-            region_results <- rbind(region_results, result)
-          }
-        }
-        # print(head(region_results))
-      }
-      if (!dir.exists(file.path(AGML_ROOT, "R-output",
-                                crop, region, indicator))) {
-        dir.create(file.path(AGML_ROOT, "R-output",
-                             crop, region, indicator),
-                  recursive=TRUE)
+        found_raster_file <- length(file_list) > 0
       }
-      write.csv(region_results,
-                file.path(AGML_ROOT, "R-output",
-                          crop, region, indicator,
-                          paste0(indicator, "_", region, ".csv")),
-                row.names=FALSE)
+
+      # resample
+      ind_rast <- rast(file_list[[1]])
+      crop_mask <- resample(crop_mask, ind_rast, method="bilinear")
+      crop_mask <- crop(crop_mask, sel_shapes)
+      resampled_filename <- file.path(AGML_ROOT, "crop_masks", "resampled_crop_mask.tif")
+      writeRaster(crop_mask, resampled_filename)
+
+      dfs <- pblapply(start_year:end_year, process_ts_year,
+                      region=region,
+                      file_path=indicator_path,
+                      filename_pattern=filename_pattern,
+                      crop_mask_filename=resampled_filename,
+                      cl=num_cpus)
+  
+      result <- rbindlist(dfs)
+      rm(list=c("ind_rast", "crop_mask", "dfs"))
```

Comment:
You probably want to remove the first two before the `pblapply`, otherwise all subprocesses will have them in memory. You can do multiple `rm`s throughout the code for anything you don't need any more